libs/datasets/timeseries.py

```diff
@@ -959,6 +960,67 @@ def drop_regions_without_population(
 LOCATION_ID_AGG = "location_id_agg"
 
 
+@dataclass(frozen=True)
+class AverageStaticWeightAggregation:
+    """Information about a timeseries field which is aggregated using a weighted average."""
+
+    # timeseries field that gets aggregated using a weighted average
+    field: FieldName
+    # static field that used to produce the weights
+    scale_factor: FieldName
+
+
+AGGREGATIONS = (
+    AverageStaticWeightAggregation(CommonFields.TEST_POSITIVITY, CommonFields.POPULATION),
+    AverageStaticWeightAggregation(CommonFields.CUMULATIVE_HOSPITALIZED, CommonFields.ICU_BEDS),
+)
+
+
+def _apply_scaling_factor(
+    timeseries: pd.DataFrame,
+    static_in: pd.DataFrame,
+    static_agg: pd.DataFrame,
+    location_id_map: Mapping[str, str],
+    aggregations: Sequence[AverageStaticWeightAggregation],
+) -> pd.DataFrame:
```

Comment:
I'm not sure this solves the problem for the hospitalization occupancy rates.  Those are only in static data and I don't see that being aggregated anywhere?